src/Renderer/ThreeExtended/Feature2Mesh.js

```diff
@@ -79,168 +84,223 @@ function coordinatesToVertices(contour, altitude, target, offset = 0, extrude =
+        geom.setIndex(new THREE.BufferAttribute(indices, 1));
+        return new THREE.LineSegments(geom, lineMaterial);
     } else {
-        return new THREE.Line(geom);
+        const color = getProperty('color', options, randomColor, feature.geometry.properties);
+        fillColorArray(colors, count, color);
+        geom.addAttribute('color', new THREE.BufferAttribute(colors, 3, true));
+        return new THREE.Line(geom, lineMaterial);
     }
 }
 
-function featureToPolygon(feature, properties, options) {
-    // get altitude / color from properties
-    const altitude = getProperty('altitude', options, 0, properties, feature.vertices);
-    const color = getProperty('color', options, randomColor, properties);
+const color = new THREE.Color();
+const material = new THREE.MeshBasicMaterial();
+function featureToPolygon(feature, options) {
+    const ptsIn = feature.vertices;
+    const normals = feature.normals;
+    const vertices = new Float32Array(ptsIn.length);
+    const colors = new Uint8Array(ptsIn.length);
+    const indices = [];
+    vertices.minAltitude = Infinity;
 
-    const geom = prepareBufferGeometry(
-        feature.vertices,
-        color,
-        altitude);
-
-    let indices = [];
     for (const geometry of feature.geometry) {
+        const altitude = getProperty('altitude', options, 0, geometry.properties);
+        const color = getProperty('color', options, randomColor, geometry.properties);
+
         const start = geometry.indices[0].offset;
         const lastIndice = geometry.indices.slice(-1)[0];
         const end = lastIndice.offset + lastIndice.count;
+        const count = end - start;
 
+        coordinatesToVertices(ptsIn, normals, vertices, altitude, 0, start, count);
+        fillColorArray(colors, count, color, start);
+
+        const geomVertices = vertices.slice(start * 3, end * 3);
         const holesOffsets = geometry.indices.map(i => i.offset - start).slice(1);
+        const triangles = Earcut(geomVertices, holesOffsets, 3);
 
-        const triangles = Earcut(geom.attributes.position.array.slice(start * 3, end * 3),
-                holesOffsets, 3);
+        const startIndice = indices.length;
+        indices.length += triangles.length;
 
-        indices = indices.concat(triangles.map(i => i + start));
+        for (let i = 0; i < triangles.length; i++) {
+            indices[startIndice + i] = triangles[i] + start;
+        }
     }
 
+    const geom = new THREE.BufferGeometry();
+    geom.addAttribute('position', new THREE.BufferAttribute(vertices, 3));
+    geom.addAttribute('color', new THREE.BufferAttribute(colors, 3, true));
+
     geom.setIndex(new THREE.BufferAttribute(new Uint16Array(indices), 1));
-    return new THREE.Mesh(geom);
+
+    const mesh = new THREE.Mesh(geom, material);
+    mesh.minAltitude = vertices.minAltitude;
+    return mesh;
 }
 
+function area(contour, offset, count) {
+    // Fix me: not the same result with THREE.shapeUtils.area
```

Comment:
This needs to be fixed before merging

Reply:
sorry, it was old comment, it was already fixed